// @tokens-studio/sd-transforms/dist/bundled_CJS_deps/postcss-calc-ast-parser/dist/index.js
import { __exports as dist } from '../../../_virtual/index.js';
import './types/ast/index.js';
import './css-calc/index.js';
import { __exports as ast } from '../../../_virtual/index2.js';
import { __exports as cssCalc } from '../../../_virtual/index3.js';

Object.defineProperty(dist, "__esModule", { value: true });
const AST = ast;
dist.AST = AST;
const css_calc_1 = cssCalc;
dist.Parser = css_calc_1.Parser;
dist.Tokenizer = css_calc_1.Tokenizer;
dist.Stringifier = css_calc_1.Stringifier;
dist.getResolvedType = css_calc_1.getResolvedType;
var reduceExpression = dist.reduceExpression = css_calc_1.reduceExpression;
dist.mathExpr = css_calc_1.newMathExpression;
function parse(code, options) {
    const tokenizer = new css_calc_1.Tokenizer(code, options);
    return new css_calc_1.Parser(tokenizer, options).parse();
}
var parse_1 = dist.parse = parse;
function stringify(node, options) {
    const stringifier = new css_calc_1.Stringifier(options);
    return stringifier.stringify(node);
}
dist.stringify = stringify;
dist.default = {
    parse,
    stringify,
    getResolvedType: css_calc_1.getResolvedType,
    reduceExpression: css_calc_1.reduceExpression,
    mathExpr: css_calc_1.newMathExpression,
    Parser: css_calc_1.Parser,
    Tokenizer: css_calc_1.Tokenizer,
    Stringifier: css_calc_1.Stringifier,
    AST,
};

export { parse_1 as parse, reduceExpression };
