// @tokens-studio/types/dist/types/index.cjs
'use strict';

var BoxShadowTypes = require('../constants/BoxShadowTypes.cjs');
var ColorModifierTypes = require('../constants/ColorModifierTypes.cjs');
var ColorSpaceTypes = require('../constants/ColorSpaceTypes.cjs');
var Properties = require('../constants/Properties.cjs');
var TokenSetStatus = require('../constants/TokenSetStatus.cjs');
var TokenTypes = require('../constants/TokenTypes.cjs');
var BorderValues = require('../constants/BorderValues.cjs');
var BoxShadowValues = require('../constants/BoxShadowValues.cjs');
var TypographyValues = require('../constants/TypographyValues.cjs');



Object.defineProperty(exports, 'BoxShadowTypes', {
	enumerable: true,
	get: function () { return BoxShadowTypes.BoxShadowTypes; }
});
Object.defineProperty(exports, 'ColorModifierTypes', {
	enumerable: true,
	get: function () { return ColorModifierTypes.ColorModifierTypes; }
});
Object.defineProperty(exports, 'ColorSpaceTypes', {
	enumerable: true,
	get: function () { return ColorSpaceTypes.ColorSpaceTypes; }
});
Object.defineProperty(exports, 'Properties', {
	enumerable: true,
	get: function () { return Properties.Properties; }
});
Object.defineProperty(exports, 'TokenSetStatus', {
	enumerable: true,
	get: function () { return TokenSetStatus.TokenSetStatus; }
});
Object.defineProperty(exports, 'TokenTypes', {
	enumerable: true,
	get: function () { return TokenTypes.TokenTypes; }
});
Object.defineProperty(exports, 'BorderValues', {
	enumerable: true,
	get: function () { return BorderValues.BorderValues; }
});
Object.defineProperty(exports, 'BoxShadowValues', {
	enumerable: true,
	get: function () { return BoxShadowValues.BoxShadowValues; }
});
Object.defineProperty(exports, 'TypographyValues', {
	enumerable: true,
	get: function () { return TypographyValues.TypographyValues; }
});
